pkg/agent/cniserver/ipam/antrea_ipam_controller.go

```diff
@@ -204,16 +193,58 @@ ownerReferenceLoop:
 	return strings.Split(annotations, annotation.AntreaIPAMAnnotationDelimiter), ips, reservedOwner, ipErr
 }
 
+// Look up IPPools from the Pod annotation.
 func (c *AntreaIPAMController) getPoolAllocatorByPod(namespace, podName string) (mineType, *poolallocator.IPPoolAllocator, []net.IP, *crdv1a2.IPAddressOwner, error) {
 	poolNames, ips, reservedOwner, err := c.getIPPoolsByPod(namespace, podName)
 	if err != nil {
 		return mineUnknown, nil, nil, nil, err
-	} else if len(poolNames) < 1 {
+	} else if len(poolNames) == 0 {
 		return mineFalse, nil, nil, nil, nil
 	}
-	// Only one pool is supported as of today
-	// TODO - support a pool for each IP version
-	ipPool := poolNames[0]
-	allocator, err := poolallocator.NewIPPoolAllocator(ipPool, c.crdClient, c.ipPoolLister)
+
+	var allocator *poolallocator.IPPoolAllocator
+	for _, p := range poolNames {
+		allocator, err = poolallocator.NewIPPoolAllocator(p, c.crdClient, c.ipPoolLister)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				err = fmt.Errorf("failed to get IPPool %s: %v", p, err)
+				break
+			}
+			klog.InfoS("IPPool not found", "pool", p)
+			err = nil
+		} else if allocator.IPVersion == crdv1a2.IPv4 {
+			// Support IPv6 / dual stack in future.
+			break
+		}
+	}
+	if allocator == nil {
+		err = fmt.Errorf("no valid IPPool found")
+	}
+
 	return mineTrue, allocator, ips, reservedOwner, err
 }
+
+// Look up IPPools by matching PodOwnder.
+func (c *AntreaIPAMController) getPoolAllocatorsByOwner(podOwner *crdv1a2.PodOwner) ([]*poolallocator.IPPoolAllocator, error) {
+	var allocators []*poolallocator.IPPoolAllocator
+	ipPools, _ := c.ipPoolInformer.Informer().GetIndexer().ByIndex(podIndex,
+		k8s.NamespacedName(podOwner.Namespace, podOwner.Name))
+	for _, item := range ipPools {
+		ipPool := item.(*crdv1a2.IPPool)
+		for _, IPAddress := range ipPool.Status.IPAddresses {
```

Comment:
I think following the convention it should be ipAddress here

Reply:
You are right. The code was copied from getIPPoolsByPod(), but let me fix it in the next PR.